### PR TITLE
refactor: event types kbar action

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?

- [x] removed early return of null (which introduced flash in some pages) https://github.com/calcom/cal.com/blob/0ec71e52eff2ac8bdfd4bdd4616c6704f9f5aaeb/packages/features/kbar/Kbar.tsx#L56
- [x] used `useRegisterActions` hook to register the commands dynamically
- [x] memoize actions
 

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)

### CODE REVIEW

**Review by zomars — APPROVED** [review]
Tested locally and is working nicely! Thanks again @G3root  🙏🏽

File: packages/features/kbar/Kbar.tsx [hunk 1]
```diff
@@ -7,14 +7,17 @@ import {
...
+  useRegisterActions,
 } from "kbar";
+import type { Action } from "kbar";
 import { useRouter } from "next/router";
 import { useMemo } from "react";
 
 import { appStoreMetadata } from "@calcom/app-store/appStoreMetaData";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import { isMac } from "@calcom/lib/isMac";
-import { RouterOutputs, trpc } from "@calcom/trpc/react";
```
Comment (alishaz-polymath) [inline packages/features/kbar/Kbar.tsx:17]: Is there a reason why this import is broken into two? 🤔

**Review by alishaz-polymath — APPROVED** [review]
Looks good. Just a minor NIT. Awesome revamp @G3root 🙌

File: packages/features/kbar/Kbar.tsx (hunk 1 above)
Comment (G3root) [reply packages/features/kbar/Kbar.tsx:17]: Due to our recent changes to eslint. we have to explicitly import as a type while importing a type.

**Comment by PeerRich** [comment]
@zomars @alishaz-polymath PR not merging, checks are failing.

lets make sure to get things merged after review ASAP

**Comment by PeerRich** [comment]
@G3root somehow checks and tests are failing, can you investigate?

**Comment by PeerRich** [comment]
hey @G3root something is not passing tests here. I think its related to this PR, can you check locally? try running yarn build

**Comment by G3root** [comment]
> hey @G3root something is not passing tests here. I think its related to this PR, can you check locally? try running yarn build

sure

File: packages/features/ee/sso/page/user-sso-view.tsx [hunk 2]
```diff
@@ -11,9 +12,11 @@ const SAMLSSO = () => {
   const { t } = useLocale();
   const router = useRouter();
 
-  if (HOSTED_CAL_FEATURES) {
-    router.push("/404");
-  }
+  useEffect(() => {
```
Comment (G3root) [inline packages/features/ee/sso/page/user-sso-view.tsx:15]: fixes https://nextjs.org/docs/messages/no-router-instance

File: apps/web/pages/settings/security/password.tsx [hunk 3]
```diff
@@ -21,7 +21,8 @@ const PasswordView = () => {
   const { t } = useLocale();
   const utils = trpc.useContext();
   const { data: user } = trpc.viewer.me.useQuery();
-  const metadata = userMetadata.parse(user?.metadata);
+  const metadata = userMetadata.safeParse(user?.metadata);
```
Comment (G3root) [inline apps/web/pages/settings/security/password.tsx:24]: fixes zod error while building for production

File: packages/features/ee/sso/page/teams-sso-view.tsx [hunk 4]
```diff
@@ -28,6 +25,12 @@ const SAMLSSO = () => {
     }
   );
 
+  useEffect(() => {
```
Comment (G3root) [inline packages/features/ee/sso/page/teams-sso-view.tsx:28]: fixes https://nextjs.org/docs/messages/no-router-instance

**Pull request merged**